src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc

```diff
@@ -1,93 +1,218 @@
+    std::promise<rpc::AddJobReply> reply_promise;
+    auto reply_future = reply_promise.get_future();
+    client_->AddJob(
+        request, [&reply_promise](const Status &status, const rpc::AddJobReply &reply) {
+          if (status.ok()) {
+            reply_promise.set_value(reply);
+          }
+        });
+    auto future_status = reply_future.wait_for(std::chrono::milliseconds(200));
+    ASSERT_EQ(future_status, std::future_status::ready);
+  }
+
+  void TestMarkJobFinished(const rpc::MarkJobFinishedRequest &request) {
+    std::promise<rpc::MarkJobFinishedReply> reply_promise;
+    auto reply_future = reply_promise.get_future();
+    client_->MarkJobFinished(
+        request,
+        [&reply_promise](const Status &status, const rpc::MarkJobFinishedReply &reply) {
+          if (status.ok()) {
+            reply_promise.set_value(reply);
+          }
+        });
+    auto future_status = reply_future.wait_for(std::chrono::milliseconds(200));
+    ASSERT_EQ(future_status, std::future_status::ready);
+  }
+
+  void TestRegisterActorInfo(const rpc::RegisterActorInfoRequest &request) {
+    std::promise<rpc::RegisterActorInfoReply> reply_promise;
+    auto reply_future = reply_promise.get_future();
+    client_->RegisterActorInfo(
+        request,
+        [&reply_promise](const Status &status, const rpc::RegisterActorInfoReply &reply) {
+          if (status.ok()) {
+            reply_promise.set_value(reply);
+          }
+        });
+    auto future_status = reply_future.wait_for(std::chrono::milliseconds(200));
+    ASSERT_EQ(future_status, std::future_status::ready);
+  }
+
+  void TestUpdateActorInfo(const rpc::UpdateActorInfoRequest &request) {
+    std::promise<rpc::UpdateActorInfoReply> reply_promise;
+    auto reply_future = reply_promise.get_future();
+    client_->UpdateActorInfo(
+        request,
+        [&reply_promise](const Status &status, const rpc::UpdateActorInfoReply &reply) {
+          if (status.ok()) {
+            reply_promise.set_value(reply);
+          }
+        });
+    auto future_status = reply_future.wait_for(std::chrono::milliseconds(200));
+    ASSERT_EQ(future_status, std::future_status::ready);
+  }
+
+  void TestGetActorInfo(const rpc::ActorTableData &expected) {
+    rpc::GetActorInfoRequest request;
+    request.set_actor_id(expected.actor_id());
+    std::promise<rpc::GetActorInfoReply> reply_promise;
+    auto reply_future = reply_promise.get_future();
+    client_->GetActorInfo(
+        request, [expected, &reply_promise](const Status &status,
+                                            const rpc::GetActorInfoReply &reply) {
+          if (status.ok()) {
+            reply_promise.set_value(reply);
+            ASSERT_TRUE(reply.actor_table_data().state() == expected.state());
+          }
+        });
+    auto future_status = reply_future.wait_for(std::chrono::milliseconds(200));
+    ASSERT_EQ(future_status, std::future_status::ready);
```

Comment:
The code for dealing with future and promise is duplicated in each function. Maybe define a helper function?

Reply:
fixed